Marlin/src/inc/Conditionals_post.h

```diff
@@ -2946,7 +2946,11 @@
 /**
  * Bed Probe dependencies
  */
-#if ANY(MESH_BED_LEVELING, HAS_BED_PROBE)
+#if ANY(BABYSTEPPING, PROBE_SELECTED)
+  #define HAS_ZOFFSET_ITEM 1
```

Comment:
Different UIs are going to have different criteria for whether or not they display a "Z Offset" item, so this flag should be left with each individual UI that currently uses it.

Reply:
that may be true, however the only UIs which have `HAS_ZOFFSET_ITEM` are exactly those three. that I am able to find.

Reply:
Yep, and all those UIs are externally and independently developed, no guarantee they'll remain in sync. So these flags should be left with their respective UIs.